coherent_rates/system.py

```diff
@@ -275,3 +333,161 @@ def get_isf(
+        state_evolved_scattered,
+    )
+
+
+def get_random_boltzmann_state(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    temperature: float,
+) -> StateVector[Any]:
+    """Generate a random Boltzmann state.
+
+    Follows the formula described in eqn 5 in
+    https://doi.org/10.48550/arXiv.2002.12035.
+
+
+    Args:
+    ----
+        system (PeriodicSystem): system
+        config (PeriodicSystemConfig): config
+        temperature (float): temperature of the system
+
+    Returns:
+    -------
+        StateVector[Any]: state with boltzmann distribution
+
+    """
+    hamiltonian = get_hamiltonian(system, config)
+    boltzmann_distribution = np.exp(
+        -hamiltonian["data"] / (2 * k * temperature),
+    )
+
+    random_phase = np.exp(2j * np.pi * np.random.rand(len(hamiltonian["data"])))
+    normalization = np.sqrt(sum(np.square(boltzmann_distribution)))
+    boltzmann_state = boltzmann_distribution * random_phase / normalization
+    return {"basis": hamiltonian["basis"][0], "data": boltzmann_state}
+
+
+def get_boltzmann_state(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    temperature: float,
+    phase: float,
+) -> StateVector[Any]:
+    hamiltonian = get_hamiltonian(system, config)
+    boltzmann_distribution = np.exp(
+        -hamiltonian["data"] / (2 * k * temperature),
+    )
+
+    normalization = np.sqrt(sum(np.square(boltzmann_distribution)))
+    boltzmann_state = boltzmann_distribution * np.exp(1j * phase) / normalization
+    return {"basis": hamiltonian["basis"][0], "data": boltzmann_state}
+
+
+def get_average_boltzmann_isf(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    direction: tuple[int] = (1,),
+    temperature: float = 300,
+    n: int = 10,
+) -> ValueList[_AX0Inv]:
+    isf_data = np.zeros(times.n, dtype=np.complex128)
+    hamiltonian = get_hamiltonian(system, config)
+    operator = get_periodic_x_operator(hamiltonian["basis"][0], direction)
+
+    for _i in range(n):
+        isf_data += get_isf_from_hamiltonian(
+            hamiltonian,
+            operator,
+            get_random_boltzmann_state(system, config, temperature),
+            times,
+        )["data"]
+    isf_data = isf_data / n
+    return {
+        "data": isf_data,
+        "basis": times,
+    }
+
+
+def get_cl_operator(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    temperature: float,
+) -> SingleBasisOperator[Any]:
+    """Generate the operator for the stationary Caldeira-Leggett solution.
+
+    Follows the formula from eqn 3.421 in
+    https://doi.org/10.1093/acprof:oso/9780199213900.001.0001
+
+    Args:
+    ----
+        system (PeriodicSystem): _description_
+        config (PeriodicSystemConfig): _description_
+        temperature (float): _description_
+
+    Returns:
+    -------
+        SingleBasisOperator[Any]: _description_
+
+    """
+    potential = get_extended_interpolated_potential(
```

Comment:
it would be good if you could write the formula as a comment here - It's hard for me to understand the code as is ...